src/core/search/vector_utils.cc

```diff
@@ -16,10 +17,21 @@ using namespace std;
 namespace {
 
 #ifdef USE_SIMSIMD
-// Native float16 support is controlled via CMake option SIMSIMD_NATIVE_F16
-// Default behavior disables native F16/BF16 for maximum compatibility
 #include <simsimd/simsimd.h>
-#endif
+#ifdef SIMSIMD_DYNAMIC_DISPATCH
+// Initialize dynamic dispatch on first use (only when dynamic dispatch is enabled)
+static bool InitializeSimSIMD() {
+  static std::once_flag simsimd_once;
+  std::call_once(simsimd_once, []() {
```

Comment:
I do not think we need `std::call_once` if you already use `static const bool` initializer below.

(btw, I am not a fan of implicit initializers, and prefer explicit ones, say called from `Service::Init`